ci/unit_tests/workspace_delete/main_test.py

```diff
@@ -0,0 +1,254 @@
+        BaseTestCaseCapture.checkEnvironmentVariables(['WA_USERNAME', 'WA_PASSWORD'])
+        cls.username = os.environ['WA_USERNAME']
+        cls.password = os.environ['WA_PASSWORD']
+
+        cls.deployParamsBase = ['-of', cls.dataBasePath, '-ow', cls.jsonWorkspaceFilename,\
+         '-cn', cls.username, '-cp', cls.password, '-cu', cls.workspacesUrl, '-cv', cls.version, '-v']
+
+        cls.deleteParamsBase = [\
+         '-cn', cls.username, '-cp', cls.password, '-cu', cls.workspacesUrl, '-cv', cls.version, '-v']
+
+    def callfunc(self, *args, **kwargs):
+        workspace_delete.main(*args, **kwargs)
+
+    def setup_method(self):
+        self._deleteAllWorkspaces()
+
+    def teardown_method(self):
+        self._deleteAllWorkspaces()
+
+    def _deleteAllWorkspaces(self):
+
+        workspaces = getWorkspaces(self.workspacesUrl, self.version, self.username, self.password)
+
+        for workspace in workspaces:
+            requestUrl = self.workspacesUrl + '/' + workspace['workspace_id'] + '?version=' + self.version
+            response = requests.delete(requestUrl, auth=(self.username, self.password), headers={'Accept': 'text/html'})
+
+        workspaces = getWorkspaces(self.workspacesUrl, self.version, self.username, self.password)
+
+        assert len(workspaces) == 0
+
+    @pytest.mark.parametrize('envVarNameUsername, envVarNamePassword', [('WA_USERNAME', 'WA_PASSWORD')])
+    def test_deleteById(self, envVarNameUsername, envVarNamePassword):
+        """Tests if workspace can be deleted by its id."""
+
+        # use outputPath instead of dataBasePath when workspace_deploy script will be able to take workspace
+        # and config file from different directories (workspace should be taken from
+        # dataBasePath and config should be saved to outputs directory)
+        createOutputConfigFilename = 'createWorkspaceOutput.cfg'
+        createOutputConfigPath = os.path.abspath(os.path.join(self.dataBasePath, createOutputConfigFilename))
+        deleteOutputConfigFilename = 'deleteWorkspaceOutput.cfg'
+        deleteOutputConfigPath = os.path.abspath(os.path.join(self.dataBasePath, deleteOutputConfigFilename))
+
+        # deploy test workspace
+        deployParams = list(self.deployParamsBase)
+        deployParams.extend(['-oc', createOutputConfigPath, '-wn', '2 workspaces with the same name'])
+        workspace_deploy.main(deployParams)
+
+        # deploy one more workspace
+        deployParamsMore = list(self.deployParamsBase)
+        deployParamsMore.extend(['-wn', '2 workspaces with the same name'])
+        workspace_deploy.main(deployParamsMore)
+
+        # try to delete workspace by its id (id is obtained from output config of deploy script)
+        deleteParams = list(self.deleteParamsBase)
+        deleteParams.extend(['-c', createOutputConfigPath, '-oc', deleteOutputConfigPath])
+        self.t_noException([deleteParams])
+
+        parser = argparse.ArgumentParser()
```

Comment:
I think it would be easier to write simple parse for obtaining workspace ID

Reply:
I somehow simplified it in b908224027fbce77cbcfeb534f1a5f76d08e7436.